chord_metadata_service/chord/workflows/wdls/experiments_json_with_files.wdl

```diff
@@ -0,0 +1,285 @@
+        Array[String] consolidated_paths_for_drs = prepare_for_drs.consolidated_paths_for_drs
+        Array[String] drs_responses = post_to_drs.response_message
+        File results_post_drs = write_drs_responses_to_file.results_post_drs
+        File processed_drs_responses = parse_txt.processed_drs_responses
+        File final_updated_json = update_experiment_json.final_updated_json
+    }
+}
+
+task prepare_files_list {
+    input {
+        File json_document
+        String directory
+    }
+    command <<<
+    python3 -c "
+import json
+import os
+
+directory = '~{directory}'
+
+with open('~{json_document}', 'r') as file:
+    data = json.load(file)
+
+path_list = []
+for experiment in data.get('experiments', []):
+    for result in experiment.get('experiment_results', []):
+        filename = result.get('filename', '')
+        file_path = ''
+        file_found = False
+
+        for root, dirs, files in os.walk(directory):
+            if filename in files:
+                file_found = True
+                file_path = os.path.join(root, filename)
+                path_list.append({'filename': filename, 'url': file_path})
+                break
+        if not file_found:
+            print(f'File not found for {filename}')
+
+with open('path_list.json', 'w') as file:
+    json.dump(path_list, file, indent=4)
+    "
+    >>>
+    output {
+        File path_list = "path_list.json"
+    }
+}
+
+task prepare_for_drs {
+    input {
+        File json_path_list
+    }
+
+    command <<<
+    python3 -c "
+import json
+
+with open('~{json_path_list}', 'r') as file:
+    path_list = json.load(file)
+
+consolidated_paths = [str(path['url']).strip() for path in path_list if str(path['url']).strip()]
+
+print(json.dumps(consolidated_paths))
+
+    "
+    >>>
+    output {
+        Array[String] consolidated_paths_for_drs = read_json(stdout())
+    }
+}
+
+task post_to_drs {
+    input {
+        File file_path
+        String drs_url
+        String project_dataset
+        String token
+        Boolean validate_ssl
+    }
+    command <<<
+        project_id=$(python3 -c 'print("~{project_dataset}".split(":")[0])')
+        dataset_id=$(python3 -c 'print("~{project_dataset}".split(":")[1])')
+        curl ~{true="" false="-k" validate_ssl} \
+            -X POST \
+            -F "file=@~{file_path}" \
+            -F "project_id=$project_id" \
+            -F "dataset_id=$dataset_id" \
+            -H "Authorization: Bearer ~{token}" \
+            --fail-with-body \
+            "~{drs_url}/ingest"
+    >>>
+    output {
+        String response_message = read_string(stdout())
+    }
+}
+
+task write_drs_responses_to_file {
+    input {
+        Array[String] drs_responses
+    }
+
+    command <<<
+        for response in "~{sep=' ' drs_responses}"; do
+            echo "$response"
+        done > results_post_drs.txt
+    >>>
+
+    output {
+        File results_post_drs = "results_post_drs.txt"
+    }
+}
```

Comment:
Output file should be a .json containing an array of responses.
The following command might work for you:
```python
import json
responses = []
for response_string in ~{drs_responses}:
   response = json.loads(response_string)
   responses.append(response)
with open('responses_list.json', 'w') as file:
    json.dump(responses, file, indent=4)
```

Reply:
Good idea! Thanks